### PR TITLE
fix(parental-leave): Refactor state machine and update documentation 

### PR DESCRIPTION
**Pull request by ylfahfa**
[UUFV-410 Refactor State Machine](https://dit-iceland.atlassian.net/browse/UUFV-410)

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Updated the Parental Leave application process to streamline user interactions and requirements.

- **Documentation**
  - Enhanced the README documentation to include detailed actions required from various stakeholders (other parent, employer, VMST) and clarified document requirements.

- **Refactor**
  - Removed outdated states and conditions from the Parental Leave service and related components, improving the logic flow and maintenance.

@coderabbitai ignore

### CODE REVIEW

File: libs/application/templates/parental-leave/README.md
Comment (veronikasif) [inline libs/application/templates/parental-leave/README.md:1]: Wrong link to flow chart in README.md file

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts
Comment (veronikasif) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:1]: I think we should add a `triggerEvent` to `onExit` in `EDIT_OR_ADD_EMPLOYERS_AND_PERIODS`. 
Should only call `validateApplication` on `SUBMIT`, not on `ABORT`

Comment (veronikasif) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:1]: Is the `setAllowanceToOtherParent` action still used? I think it's not needed anymore (we have updated the answers obj)

Comment (ylfahfa) [reply libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:1]: I don't know anything about `setAllowanceToOtherParent` but I have removed it as you say it is not needed anymore 😁

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 1]
```diff
@@ -877,15 +804,8 @@ const ParentalLeaveTemplate: ApplicationTemplate<
...
-                goToState(
-                  application,
-                  States.VINNUMALASTOFNUN_APPROVAL_ABORT_CHANGE,
-                ),
-              target: States.VINNUMALASTOFNUN_APPROVAL_ABORT_CHANGE,
-            },
-            {
-              target: States.VINNUMALASTOFNUN_APPROVE_EDITS_ABORT,
+                goToState(application, States.VINNUMALASTOFNUN_APPROVAL),
+              target: States.VINNUMALASTOFNUN_APPROVAL,
```
[the remark this answers is not in the document]
Comment (ylfahfa) [reply libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:808]: Hmm, I think I did that by accident when removing `VINNUMALASTOFNUN_APPROVAL_ABORT_CHANGE`. But are we missing an arrow from VMST Approve Edits in the flow chart?

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 2]
```diff
@@ -1311,6 +1227,14 @@ const ParentalLeaveTemplate: ApplicationTemplate<
 
           onEntry: [
             defineTemplateApi({
+              triggerEvent: DefaultEvents.APPROVE,
+              action: ApiModuleActions.sendApplication,
+              params: FileType.DOCUMENTPERIOD,
```
[the remark this answers is not in the document]
Comment (ylfahfa) [reply libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:1232]: I'm not sure. Don't really know about this param? Didn't Karen also say it wasn't the right one? I forgot I was going to get input on this.

[the remark this answers is not in the document]
Comment (ylfahfa) [reply libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:1233]: Removed param from `APPROVE`

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Review by karenbjorg — APPROVED** [review]
nice work 💯

**Pull request merged**